spacy/strings.pyx

```diff
@@ -123,127 +32,126 @@ cdef class StringStore:
             for string in strings:
                 self.add(string)
 
-    def __getitem__(self, object string_or_id):
-        """Retrieve a string from a given hash, or vice versa.
+    def __getitem__(self, string_or_hash: Union[str, int]) -> Union[str, int]:
+        """Retrieve a string from a given hash. If a string
+        is passed as the input, add it to the store and return
+        its hash.
 
-        string_or_id (bytes, str or uint64): The value to encode.
-        Returns (str / uint64): The value to be retrieved.
+        string_or_hash (str, int): The hash value to lookup or the string to store.
+        RETURNS (str, int): The stored string or the hash of the newly added string.
```

Comment:
```suggestion
        string_or_hash (int / str): The hash value to lookup or the string to store.
        RETURNS (str / int): The stored string or the hash of the newly added string.
```
(keeping the order aligned with the docstring for hopefully a bit more readability)